include/sparrow/utils/ranges.hpp

```diff
@@ -92,7 +92,14 @@ namespace sparrow
             {
                 for (; first != last; ++first, (void) ++result)
                 {
+#if defined(__GNUC__)
```

Comment:
let's add ` && __GNUC__ < 12`  so we know that we can remove this when we drop support for gcc11

Reply:
ok